xrpl_types/src/currency.rs

```diff
@@ -38,35 +38,63 @@ impl Currency {
     }
 }
 
-// #TODO implement from/into.
-/// Currency specification for books.
-#[derive(Debug, Clone, Serialize)]
-pub struct CurrencySpec {
-    pub currency: String, // TODO: hm, consider name.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub issuer: Option<String>,
-}
+impl Serialize for Currency {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Debug, Clone, Serialize)]
```

Comment:
Hmm, I am wondering if this lexically-scoped stuct definition has any performance issue. Interesting approach.

Reply:
I don't think so, it is just like any other type, just limited scope. It is a quite common way to serialize stuff in a custom way without having to call methods on `Serializer` explicitly (we rely on the `Serialize` derive macro).